File: backend/contract-application/src/main/java/com/app/model/Pegawai.java
package com.app.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "pegawai")
public class Pegawai {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idPegawai;
    @Column(unique = true, nullable = false)
    private String kodePegawai;
    @Column(nullable = false)
    private String namaPegawai;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_jabatan", nullable = false)
    @JsonIgnoreProperties("pegawaiList")
    private Jabatan jabatan;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_cabang", nullable = false)
    @JsonIgnoreProperties("pegawaiList")
    private Cabang cabang;

    @Column(nullable = false)
    private LocalDate tanggalMulaiKontrak;
    @Column(nullable = false)
    private LocalDate tanggalHabisKontrak;
}

File: frontend/PegawaiFE/Models/Pegawai.cs
using System.Text.Json.Serialization;

namespace PegawaiFE.Models
{
    public class Pegawai
    {
        [JsonPropertyName("idPegawai")]
        public long IdPegawai { get; set; }

        [JsonPropertyName("kodePegawai")]
        public string KodePegawai { get; set; } = string.Empty;

        [JsonPropertyName("namaPegawai")]
        public string NamaPegawai { get; set; } = string.Empty;

        [JsonPropertyName("jabatan")]
        public Jabatan? Jabatan { get; set; }

        [JsonPropertyName("cabang")]
        public Cabang? Cabang { get; set; }

        [JsonPropertyName("tanggalMulaiKontrak")]
        public DateTime TanggalMulaiKontrak { get; set; }

        [JsonPropertyName("tanggalHabisKontrak")]
        public DateTime TanggalHabisKontrak { get; set; }
    }
}
